internal/injector/injector.go

```diff
@@ -62,17 +63,28 @@ type (
 func New(pkgDir string, opts Options) (*Injector, error) {
 	fileset := token.NewFileSet()
 	cfg := &packages.Config{
-		Dir:  opts.Dir,
-		Fset: fileset,
+		// Explicitly disable toolexec for this, as if provided via $GOFLAGS, it
+		// would be honored by the go/packages loader and that'd cause this call to
+		// become a fork-bomb.
+		BuildFlags: []string{"-toolexec="},
+		Dir:        opts.Dir,
+		Fset:       fileset,
 		Mode: packages.NeedName |
 			packages.NeedFiles |
-			packages.NeedCompiledGoFiles |
 			packages.NeedImports |
 			packages.NeedTypes |
 			packages.NeedTypesSizes |
 			packages.NeedSyntax |
 			packages.NeedTypesInfo,
 	}
+	if flags, err := goflags.Flags(); err == nil {
+		// Honor any `-tags`  flags provided by the user, as these may affect what
+		// is getting compiled or not.
+		if tags, hasTags := flags.Get("-tags"); hasTags {
+			cfg.BuildFlags = append(cfg.BuildFlags, fmt.Sprintf("-tags=%s", tags))
```

Comment:
just curious about this: did you uncover this case some of your integration tests?

Reply:
No, but I was thinking about how to design a scenario that'd be a straight-forward demonstration of this.